packages/eas-cli/src/build/build.ts

```diff
@@ -388,6 +409,47 @@ async function handleSingleBuildProgressAsync(
             )}`
           );
         }
+
+        if (
+          build.platform === AppPlatform.Ios &&
+          [BuildResourceClass.IosIntelLarge, BuildResourceClass.IosIntelMedium].includes(
+            build.resourceClass
+          )
+        ) {
+          let askToSwitchToM1 = false;
+          if (
+            build.priority === BuildPriority.High &&
+            build.estimatedWaitTimeLeftSeconds >= /* 10 minutes */ 10 * 60
+          ) {
+            Log.newLine();
+            Log.warn(
+              `Warning: Priority queue wait time for legacy iOS Intel workers is longer than usual (more than 10 minutes).`
+            );
+            Log.warn(`We recommend switching to the new, faster M1 workers for iOS builds.`);
+            Log.warn(`Learn more: https://blog.expo.dev/m1-workers-on-eas-build-dcaa2c1333ad`);
+            askToSwitchToM1 = true;
+          } else if (
+            build.priority !== BuildPriority.High &&
+            build.estimatedWaitTimeLeftSeconds >= /* 120 minutes */ 120 * 60
+          ) {
+            Log.newLine();
+            Log.warn(
+              `Warning: Free tier queue wait time for legacy iOS Intel workers is longer than usual.`
+            );
+            Log.warn(`We recommend switching to the new, faster M1 workers for iOS builds.`);
+            Log.warn(`Learn more: https://blog.expo.dev/m1-workers-on-eas-build-dcaa2c1333ad`);
+            askToSwitchToM1 = true;
+          }
+          if (!nonInteractive && askToSwitchToM1) {
+            const shouldSwitchToM1 = await confirmAsync({
+              message: `Switch iOS builds to M1 workers (modifies build profiles in eas.json)?`,
```

Comment:
1. I think it's not clear that the current build in queue will not be switched to M1
2. I suspect that the most common flow when users see this message will be hitting Ctrl-C and running new build without canceling the current one. Maybe we should prompt users if they want to cancel the current build.

Reply:
Discussed offline